src/include/lean/lean.h

```diff
@@ -1213,6 +1244,24 @@ static inline lean_obj_res lean_nat_mul(b_lean_obj_arg a1, b_lean_obj_arg a2) {
         return lean_nat_big_mul(a1, a2);
     }
 }
+#else
+static inline lean_obj_res lean_nat_mul(b_lean_obj_arg a1, b_lean_obj_arg a2) {
+    if (LEAN_LIKELY(lean_is_scalar(a1) && lean_is_scalar(a2))) {
+        ptrdiff_t n1 = (ptrdiff_t)lean_unbox(a1);
```

Comment:
Why the cast to `ptrdiff_t`?

Reply:
Sadly, ssize_t does not seem to be in C standard. 

https://pubs.opengroup.org/onlinepubs/9699919799/basedefs/sys_types.h.html

According to C++, std::ssize returns ptrdiff_t as a signed counterpart of size_t.

Please also notice that intptr_t/uintptr_t is marked as optional while ptrdiff_t is a part of standard since C99 standard (ISO/IEC 9899:1999).

Reply:
Why do you need a signed type here? `__builtin_mul_overflow` appears to work for unsigned multplication too, as far as I can tell.

Reply:
We are multiplying two unsigned integers. If result spans to last sign bit, then it must be the case where a plain box cannot encode the whole integer result.

Reply:
For example, the following function has different behaviors:
```
#include <cstddef>
#include <iostream>
#include <cstdint>
bool f(size_t i, size_t j) {
    ptrdiff_t res;
    return __builtin_mul_overflow((ptrdiff_t)i, (ptrdiff_t)j, &res);
}

bool g(size_t i, size_t j) {
    size_t res;
    return __builtin_mul_overflow((size_t)i, (size_t)j, &res);
}

int main() {
    std::cout << std::boolalpha << f(SIZE_MAX/2, 2) << " " << g(SIZE_MAX/2, 2)  << std::endl;
}
```
Consider multiplying
```
lean_box(SIZE_MAX/2), lean_box(2)
```
the result will not be able to be stored in `lean_box`. Signed `mul` will detect this correctly.